### ISSUE TITLE
Qwen3-vl-4b-thinking + GRPO + geo3k会训崩么

### ISSUE THREAD

**Issue by xuhang-2** [verl-project/verl#4818]
请问各位大佬，上述setting下发现reward曲线会突然drop, 类似训崩了

<img width="461" height="300" alt="Image" src="https://github.com/user-attachments/assets/c555ca15-bc21-4dc5-a819-067fb3dde3eb" />

以下是我的.sh
`
export CUDA_DEVICE_MAX_CONNECTIONS=1 # For megatron communication/computation overlapping

HF_MODEL_PATH=xx/model/Qwen3-VL-4B-Thinking/hf
train_path=xx/geo3k/train.parquet
test_path=xxgeo3k/test.parquet

OUTPUT_DIR=xx/output
mkdir -p "$OUTPUT_DIR"
project_name=verl_grpo_example_geo3k
experiment_name=qwen3_vl_4b_megatron
CKPTS_DIR=${OUTPUT_DIR}/ckpts/"${project_name}"/"${experiment_name}"
LOG_DIR=${OUTPUT_DIR}/logs/$(date +%Y%m%d_%H%M%S)

train_batch_size=64
tensor_model_parallel_size=4
rollout_N=8
nnodes=4

export VLLM_ALLREDUCE_USE_SYMM_MEM=0 # for vllm0.11.0 with TP

python3 -m verl.trainer.main_ppo --config-path=config \
    --config-name='ppo_megatron_trainer.yaml'\
    algorithm.adv_estimator=grpo \
    data.train_files="$train_path" \
    data.val_files="$test_path" \
    data.train_batch_size=$train_batch_size \
    data.max_prompt_length=1024 \
    data.max_response_length=10240 \
    data.filter_overlong_prompts=True \
    data.truncation='error' \
    actor_rollout_ref.model.path=$HF_MODEL_PATH \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.actor.ppo_mini_batch_size=$train_batch_size \
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=16 \
    actor_rollout_ref.actor.megatron.pipeline_model_parallel_size=1 \
    actor_rollout_ref.actor.megatron.tensor_model_parallel_size=$tensor_model_parallel_size \
    actor_rollout_ref.actor.megatron.context_parallel_size=1 \
    actor_rollout_ref.actor.use_kl_loss=False \
    actor_rollout_ref.actor.kl_loss_coef=0 \
    actor_rollout_ref.actor.kl_loss_type=low_var_kl \
    actor_rollout_ref.actor.entropy_coeff=0 \
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=1 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=$tensor_model_parallel_size \
    actor_rollout_ref.actor.use_dynamic_bsz=True \
    actor_rollout_ref.actor.ppo_max_token_len_per_gpu=81960 \
    actor_rollout_ref.ref.log_prob_use_dynamic_bsz=True \
    actor_rollout_ref.ref.log_prob_max_token_len_per_gpu=81960 \
    actor_rollout_ref.rollout.log_prob_use_dynamic_bsz=True \
    actor_rollout_ref.rollout.log_prob_max_token_len_per_gpu=81960 \
    actor_rollout_ref.rollout.name=vllm \
    +actor_rollout_ref.rollout.engine_kwargs.vllm.disable_mm_preprocessor_cache=True \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.8 \
    actor_rollout_ref.rollout.n=$rollout_N \
    actor_rollout_ref.rollout.max_num_batched_tokens=40960 \
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=1 \
    actor_rollout_ref.actor.megatron.use_mbridge=True \
    actor_rollout_ref.actor.megatron.param_offload=False \
    actor_rollout_ref.actor.megatron.optimizer_offload=False \
    actor_rollout_ref.actor.megatron.grad_offload=False \
    actor_rollout_ref.ref.megatron.param_offload=False \
    +actor_rollout_ref.actor.optim.override_optimizer_config.optimizer_offload_fraction=0 \
    +actor_rollout_ref.actor.optim.override_optimizer_config.overlap_cpu_optimizer_d2h_h2d=False \
    +actor_rollout_ref.actor.optim.override_optimizer_config.use_precision_aware_optimizer=True \
    +actor_rollout_ref.actor.optim.override_optimizer_config.optimizer_cpu_offload=False \
    +actor_rollout_ref.actor.megatron.override_transformer_config.moe_router_dtype=fp32 \
    +actor_rollout_ref.actor.megatron.override_transformer_config.moe_enable_deepep=True \
    +actor_rollout_ref.actor.megatron.override_transformer_config.moe_token_dispatcher_type=flex \
    +actor_rollout_ref.actor.megatron.override_transformer_config.recompute_method=uniform \
    +actor_rollout_ref.actor.megatron.override_transformer_config.recompute_granularity=full \
    +actor_rollout_ref.actor.megatron.override_transformer_config.recompute_num_layers=1 \
    +actor_rollout_ref.actor.megatron.override_transformer_config.gradient_accumulation_fusion=True \
    +actor_rollout_ref.actor.megatron.override_transformer_config.moe_permute_fusion=True \
    algorithm.use_kl_in_reward=False \
    trainer.critic_warmup=0 \
    trainer.logger='["console","swanlab"]' \
    trainer.project_name=$project_name \
    trainer.experiment_name=$experiment_name \
    trainer.n_gpus_per_node=8 \
    trainer.nnodes=$nnodes \
    trainer.val_before_train=True \
    trainer.save_freq=400 \
    trainer.test_freq=5 \
    trainer.default_local_dir="${CKPTS_DIR}" \
    trainer.total_epochs=15 $@ 2>&1 | tee "$LOG_DIR/train.log"

bash stop_ray.sh
`

**Comment by zhihaofang1017** [comment]
请问这个问题有解决吗，我在qwen3-vl-30B-A3B的模型上也遇到了这个问题

**Comment by xuhang-2** [comment]
> 请问这个问题有解决吗，我在qwen3-vl-30B-A3B的模型上也遇到了这个问题

加KL+调低学习率+增加response_length

**Comment by zhihaofang1017** [comment]
> > 请问这个问题有解决吗，我在qwen3-vl-30B-A3B的模型上也遇到了这个问题
> 
> 加KL+调低学习率+增加response_length

大佬方便提供一下现在训练的脚本嘛

**Comment by z-x-x136** [comment]
> > 请问这个问题有解决吗，我在qwen3-vl-30B-A3B的模型上也遇到了这个问题
> 
> 加KL+调节低学习率+增加response_length

还有其他办法吗，奖励坍塌了，是不是模型太小了

**Comment by Kirrito-k423** [comment]
中文：从当前信息看，这更像 GRPO 训练稳定性和配置问题，还不能判定为 verl 代码 bug。你给的脚本里同时关闭了 actor KL（`actor_rollout_ref.actor.use_kl_loss=False`、`kl_loss_coef=0`）和 reward KL（`algorithm.use_kl_in_reward=False`），而当前 GRPO 文档建议 GRPO 使用 actor KL loss，Qwen3-VL 的 geo3k 示例也默认开启 `actor_rollout_ref.actor.use_kl_loss=True` 且使用 `kl_loss_coef=0.01`。issue 里作者后续也反馈过可尝试“加 KL + 调低学习率 + 增加 response_length”。建议先基于当前 `examples/grpo_trainer/run_qwen3_vl_8b_megatron.sh` 或 `run_qwen3_vl_30b_a3b_megatron.sh` 对齐配置，至少尝试开启 actor KL、调小 LR、检查 `max_response_length` 是否会截断 `<think>...\boxed{}` 输出，并贴出 verl 版本、模型精确 checkpoint、完整 reward/kl/entropy/response length 曲线和崩溃前后的样本输出，方便进一步判断是否是实现问题。

English: Based on the available information, this looks more like a GRPO training stability/configuration issue than a confirmed verl bug. The provided script disables both actor KL (`actor_rollout_ref.actor.use_kl_loss=False`, `kl_loss_coef=0`) and reward KL (`algorithm.use_kl_in_reward=False`), while the current GRPO docs recommend actor KL loss for GRPO, and the Qwen3-VL geo3k examples enable `actor_rollout_ref.actor.use_kl_loss=True` with `kl_loss_coef=0.01` by default. The issue author also later reported that adding KL, lowering the learning rate, and increasing response length helped. Please first align with the current Qwen3-VL GRPO examples, try enabling actor KL, lowering LR, and verifying that `max_response_length` is not truncating the required `<think>...\boxed{}` output. If the collapse persists, please share the verl version, exact model checkpoint, full reward/KL/entropy/response-length curves, and sample outputs before/after the drop so maintainers can tell whether this is an implementation issue.
This response was generated by a scheduled Codex GPT5.5 issue processing task.
